include/game_options.h

```diff
@@ -0,0 +1,34 @@
+#ifndef POKEPLATINUM_GAME_OPTIONS_H
+#define POKEPLATINUM_GAME_OPTIONS_H
+
+#include "savedata.h"
+
+typedef struct Options {
+    u16 textSpeed : 4;
+    u16 soundMethod : 2;
+    u16 battleStyle : 1;
+    u16 battleScene : 1;
+    u16 buttonMode : 2;
+    u16 frame : 5;
+    u16 : 1;
+} Options;
+
+Options* Options_AllocMemory(u32 heapID);
+void Options_Copy(const Options *src, Options *dest);
+void Options_Init(Options *options);
+void Options_SetSystemButtonMode(SaveData *saveData, int buttonMode);
+int Options_TextSpeed(const Options *options);
+void Options_SetTextSpeed(Options *options, int textSpeed);
```

Comment:
For the `Set` methods, do they match if you use the `enum` types defined in `constants/game_options.h` in the signature over raw `int`?

Reply:
That does match, but I thought we weren't typedef'ing enums. Is it an exception if it's a scenario where it increase clarity to typedef it?

Reply:
What I'm suggesting is signatures like this:

```c
void Options_SetTextSpeed(Options *options, enum OptionsTextSpeed textSpeed);
```

You don't need to `typedef` the `enum` to get that functionality; the compiler will just look for an `enum` definition that matches that name.

Reply:
Ah, gotcha, sounds good. Will do, and I'll trim the parameter names a bit to make it a bit less verbose.